extensions/amp-bind/0.1/bind-impl.js

```diff
@@ -553,6 +554,33 @@ export class Bind {
     return value;
   }
 
+  /**
+   * Returns a copy of the global state for an expression, after waiting for its
+   * associated 'amp-state' element to finish fetching data. If there is no
+   * corresponding 'amp-state' element in the DOM, then reject.
+   *
+   * e.g. "foo.bar".
+   * @param {string} expr
+   * @return {!Promise<*>}
+   */
+  getStateAsync(expr) {
+    const stateKey = /^[^.]*/.exec(expr)[0];
```

Comment:
Feels as though some sufficiently smart compiler should be able to take the more readable version and automatically produce the regex, but 🤷‍♂

Reply:
I'd be surprised if the regex was faster. `indexOf` and `substr` should be even faster for longer strings.

Reply:
Forgot to mention, I actually [measured it](https://jsperf.com/split-vs-regex-stateids). Regex appears to be faster, but since we can do millions of the operation per second, I doubt it matters.